### PR TITLE
CLDR-14755 enable locales for grammar collection

### PR DESCRIPTION
**Pull request by macchiati**
CLDR-14755 enable locales for grammar collection

- [ ] This PR completes the ticket.
ALLOW_MANY_COMMITS=true

### CODE REVIEW

**Comment by macchiati** [comment]
This is a first pass to enable the grammatical data for locales we added in v39. I expect to narrow down the number of genders/cases as a result of the survey before Shakedown.

**Comment by srl295** [comment]
@macchiati i had to make a sub-PR: https://github.com/macchiati/cldr/pull/7

**Comment by srl295** [comment]
ansible lint is just a timeout, looks like it passed (in "only" 35min!)

**Review by pedberg-icu — DISMISSED** [review]

File: tools/cldr-code/src/main/java/org/unicode/cldr/util/GrammarInfo.java [hunk 1]
```diff
@@ -470,4 +476,88 @@ private void getSourceCaseAndPluralPolish(String gender, String value,
...
+    }
+
+    /**
+     * Return the locales that have either case or gender info for units (or both).
+     */
+    public static Set<String> getGrammarLocales() {
+        return GrammarLocales.data;
+    }
+
+    static final Set<String> INCLUDE_OTHER = ImmutableSet.of(
```
Comment (srl295) [inline tools/cldr-code/src/main/java/org/unicode/cldr/util/GrammarInfo.java:494]: should be data driven

**Review by srl295 — APPROVED** [review]
LG so far

**Comment by srl295** [comment]
mem fixed in https://github.com/unicode-org/cldr/pull/1259

File: tools/cldr-code/src/main/java/org/unicode/cldr/util/GrammarInfo.java (hunk 1 above)
Comment (macchiati) [reply tools/cldr-code/src/main/java/org/unicode/cldr/util/GrammarInfo.java:494]: I thought of that, but wanted to work out exactly the best way to do it.

**Review by srl295 — APPROVED** [review]
rslgtm

**Pull request merged**